fleet-charts/hello-world/templates/hpa.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2021-2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: hello-web-autoscale
+  labels:
+    app: hello
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: hello-web
+  minReplicas: 1
+  maxReplicas: 5
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 30
```

Comment:
Given that the hello world app is so trivial, will it be hard to cause enough CPU utilization to get it to scale up?

Reply:
Yeah - I'd wondered about that.  

What if we reduced the `averageUtilization` to something extremely small, like `1`?  We have a `maxReplicas` of 5, so it won't scale out of control.

Reply:
Could you test it out with a load generator and see what happens? Hopefully we can get it to do what we want.

If push comes to shove we could use a container that publishes per-request metrics then configure the HPA to scale based on the pod RPS using custom metrics. Something like this: https://github.com/brancz/prometheus-example-app (and then of course we'd have to set up stackdriver to scrape the metrics)

Reply:
Thanks @ostrain - I will circle back to this after https://github.com/GoogleCloudPlatform/gke-fleet-management/pull/17 is complete.  We could also parametrize the averageUtilization value.